fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexerSimpleTest.java

```diff
@@ -883,4 +883,88 @@ public void testConfigLoader() throws Exception {
+                ret = recordStore2.evaluateAggregateFunction(indexTypes, aggregateFunction, TupleRange.ALL, IsolationLevel.SERIALIZABLE).join();
+                // Do NOT commit changes
+            }
+            return ret;
+        };
+
+        openSimpleMetaData(hook);
+        final FDBStoreTimer timer = new FDBStoreTimer();
+
+        try (FDBRecordContext context = openContext()) {
+            recordStore.checkVersion(null, FDBRecordStoreBase.StoreExistenceCheck.ERROR_IF_NOT_EXISTS).join();
+
+            timer.reset();
+
+            // Build in this transaction.
+            try (OnlineIndexer indexer =
+                         OnlineIndexer.newBuilder()
+                                 .setRecordStore(recordStore)
+                                 .setTimer(timer)
+                                 .setIndex("newIndex")
+                                 .setLimit(100000)
+                                 .setMaxWriteLimitBytes(1)
+                                 .build()) {
+                // this call will "flatten" the staccato iterations to a whole range. Testing compatibility.
+                indexer.rebuildIndex(recordStore);
+            }
+            recordStore.markIndexReadable("newIndex").join();
+
+            assertEquals(200, timer.getCount(FDBStoreTimer.Counts.ONLINE_INDEX_BUILDER_RECORDS_SCANNED));
+            assertEquals(200, timer.getCount(FDBStoreTimer.Counts.ONLINE_INDEX_BUILDER_RECORDS_INDEXED));
+
+            assertEquals(200, timer.getCount(FDBStoreTimer.Counts.ONLINE_INDEX_BUILDER_RANGE_BY_SIZE));
+            assertEquals(0, timer.getCount(FDBStoreTimer.Counts.ONLINE_INDEX_BUILDER_RANGE_BY_COUNT));
+
+            recordStore.clearAndMarkIndexWriteOnly("newIndex").join();
+            context.commit();
+        }
+
+        try (FDBRecordContext context = openContext()) {
+            recordStore.checkVersion(null, FDBRecordStoreBase.StoreExistenceCheck.ERROR_IF_NOT_EXISTS).join();
+
+            timer.reset();
+
+            // Build in this transaction.
+            try (OnlineIndexer indexer =
+                         OnlineIndexer.newBuilder()
+                                 .setRecordStore(recordStore)
+                                 .setTimer(timer)
+                                 .setIndex("newIndex")
+                                 .setLimit(100000)
+                                 .setMaxWriteLimitBytes(1)
+                                 .build()) {
+
+                Key.Evaluated key = indexer.buildUnbuiltRange(Key.Evaluated.scalar(0L), Key.Evaluated.scalar(25L)).join();
+                assertEquals(1, key.getLong(0));
+                assertEquals(1, timer.getCount(FDBStoreTimer.Counts.ONLINE_INDEX_BUILDER_RANGE_BY_SIZE));
+                assertEquals(0, timer.getCount(FDBStoreTimer.Counts.ONLINE_INDEX_BUILDER_RANGE_BY_COUNT));
+            }
+            recordStore.clearAndMarkIndexWriteOnly("newIndex").join();
+            context.commit();
+        }
```

Comment:
I think this probably could use one more test where it calls `OnlineIndexer.buildIndex` (i.e., building the index across multiple transactions) and makes sure that we get a built index, at the end of it.

Reply:
Just pushed the thenApply (sorry, forgot about it yesterday)

I would need some guidance writing the additional test. Of all the existing indexer tests, Is there one I can use as an example?